examples/platform/efr32/BaseApplication.h

```diff
@@ -0,0 +1,172 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+/**********************************************************
+ * Includes
+ *********************************************************/
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "AppEvent.h"
+#include "FreeRTOS.h"
+#include "sl_simple_button_instances.h"
+#include "timers.h" // provides FreeRTOS timer support
+#include <app/clusters/identify-server/identify-server.h>
+#include <ble/BLEEndPoint.h>
+#include <lib/core/CHIPError.h>
+#include <platform/CHIPDeviceLayer.h>
+
+/**********************************************************
+ * Defines
+ *********************************************************/
+
+// Application-defined error codes in the CHIP_ERROR space.
+#define APP_ERROR_EVENT_QUEUE_FAILED CHIP_APPLICATION_ERROR(0x01)
+#define APP_ERROR_CREATE_TASK_FAILED CHIP_APPLICATION_ERROR(0x02)
+#define APP_ERROR_UNHANDLED_EVENT CHIP_APPLICATION_ERROR(0x03)
+#define APP_ERROR_CREATE_TIMER_FAILED CHIP_APPLICATION_ERROR(0x04)
+#define APP_ERROR_START_TIMER_FAILED CHIP_APPLICATION_ERROR(0x05)
+#define APP_ERROR_STOP_TIMER_FAILED CHIP_APPLICATION_ERROR(0x06)
+
+/**********************************************************
+ * BaseApplication Declaration
+ *********************************************************/
+
+class BaseApplication
+{
+
+public:
+    BaseApplication() = default;
+    virtual ~BaseApplication(){};
+
+    /**
+     * @brief Create AppTask task and Event Queue
+     * If an error occurs during creation, application will hang after printing out error code
+     *
+     * @return CHIP_ERROR CHIP_NO_ERROR if no errors
+     */
+    CHIP_ERROR StartAppTask(TaskFunction_t taskFunction);
+
+    /**
+     * @brief PostEvent function that add event to AppTask queue for processing
+     *
+     * @param event AppEvent to post
+     */
+    static void PostEvent(const AppEvent * event);
+
+    /**
+     * @brief Event handler when a button is pressed
+     * Function posts an event for button processing
+     *
+     * @param buttonHandle APP_LIGHT_SWITCH or APP_FUNCTION_BUTTON
+     * @param btnAction button action - SL_SIMPLE_BUTTON_PRESSED,
+     *                  SL_SIMPLE_BUTTON_RELEASED or SL_SIMPLE_BUTTON_DISABLED
+     */
+    virtual void ButtonEventHandler(const sl_button_t * buttonHandle, uint8_t btnAction) = 0;
+
+    /**
+     * @brief Function called to start the LED light timer
+     */
+    void StartLightTimer(void);
```

Comment:
the name LightTimer should probably be more generic also (and the other following LightXXX functions

Reply:
fixed